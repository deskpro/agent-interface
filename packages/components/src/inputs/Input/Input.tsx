import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/elements/dp-forms.css";
import Icon from "../../elements/Icon/Icon";

export interface InputProps
  extends React.DetailedHTMLProps<
    React.InputHTMLAttributes<HTMLInputElement>,
    HTMLInputElement
  > {
  error?: string;
  icon?: string;
}

const Input: React.FC<InputProps> = ({
  error,
  icon,
  type = "text",
  className,
  ...props
}) => (
  <div className="dp-Form-item">
    {!!icon && <Icon name={icon} size={13} />}
    <input
      type={type}
      className={classNames("dp-Input", className, `Input--${type}`, {
        "is-error": !!error
      })}
      {...props}
    />
    {!!error && <span className="dp-Input-error">{error}</span>}
  </div>
);

export default Input;
