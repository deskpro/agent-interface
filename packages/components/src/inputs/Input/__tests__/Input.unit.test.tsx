import * as React from "react";
import { render } from "react-testing-library";

import Input from "../Input";

describe("<Input />", () => {
  it("it should match its snapshot", () => {
    const handleChange = jest.fn();
    const { container } = render(
      <Input onChange={handleChange} type="text" value="Input value" />
    );
    expect(container.querySelector(".dp-Form-item")).toMatchSnapshot();
  });
});
