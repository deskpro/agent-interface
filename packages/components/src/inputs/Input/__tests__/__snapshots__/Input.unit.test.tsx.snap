// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match its snapshot 1`] = `
<div
  class="dp-Form-item"
>
  <input
    class="dp-Input Input--text"
    type="text"
    value="Input value"
  />
</div>
`;
